x-pack/plugins/session_view/public/components/ProcessTreeNode/index.tsx

```diff
@@ -39,34 +40,41 @@ export function ProcessTreeNode({
   isOrphan,
   depth = 0,
   onProcessSelected,
+  onToggleChild,
+  onToggleAlerts,
 }: ProcessDeps) {
   const textRef = useRef<HTMLSpanElement>(null);
-
-  const [childrenExpanded, setChildrenExpanded] = useState(isSessionLeader || process.autoExpand);
-  const [alertsExpanded, setAlertsExpanded] = useState(false);
   const { searchMatched } = process;
 
-  useEffect(() => {
-    setChildrenExpanded(isSessionLeader || process.autoExpand);
-  }, [isSessionLeader, process.autoExpand]);
-
   const processDetails = useMemo(() => {
     return process.getDetails();
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [process.events.length]);
+  }, [process.id]);
```

Comment:
doesn't process.id not change? if we get a fork, and then on the next page an exec and exit flow in, will this return the right data?

Reply:
as due to windowing the logic changed into a List rather than recursive nodes, whenever scrolling or loading more data, it rerenders the parent List, thus ending up re-rendering the children components. (What now makes me see that this memo optimization might not work with windowing 🤔, instead, any memoization logic should come from the parent)